src/reader/get_readings.py

```diff
@@ -168,6 +168,7 @@ def get_readout(node):
         output_fields.update(
             {
                 DEVICE_ID_KEY: OUTPUT_READINGS_DEV_ID,
+                VENDOR_ID_KEY: node.config['calc_vendor_id']
```

Comment:
This is a neat solution overall. However, it will raise an exception and result in no readout for any case where `calc_vendor_id` is not set.

I would add a separate `output_fields.update(...)` function which gets called only `if 'calc_vendor_id' in node.config`.

Finally, let's save the string `'calc_vendor_id'` in a constant, e.g. `CALC_VENDOR_ID = 'calc_vendor_id'`

Reply:
Hey @svet-b thanks for the quick feedback ! Just implemented the function. Please have a look and let me know if I can improve it further.